pkg/cloud-provider/cloudapi/azure/azure_api_wrappers.go

```diff
@@ -16,236 +16,264 @@ package azure
+	listComplete(ctx context.Context, resourceGroupName string) ([]armnetwork.ApplicationSecurityGroup, error)
+	listAllComplete(ctx context.Context) ([]armnetwork.ApplicationSecurityGroup, error)
 	delete(ctx context.Context, resourceGroupName string, applicationSecurityGroupName string) error
 }
+
 type azureAsgWrapperImpl struct {
-	asgAPIClient network.ApplicationSecurityGroupsClient
+	asgAPIClient armnetwork.ApplicationSecurityGroupsClient
 }
 
 func (asg *azureAsgWrapperImpl) createOrUpdate(ctx context.Context, resourceGroupName string,
-	applicationSecurityGroupName string, parameters network.ApplicationSecurityGroup) (network.ApplicationSecurityGroup, error) {
-	var appsg network.ApplicationSecurityGroup
+	applicationSecurityGroupName string, parameters armnetwork.ApplicationSecurityGroup) (armnetwork.ApplicationSecurityGroup, error) {
+	var appsg armnetwork.ApplicationSecurityGroup
 	asgClient := asg.asgAPIClient
-	future, err := asgClient.CreateOrUpdate(ctx, resourceGroupName, applicationSecurityGroupName, parameters)
+	poller, err := asgClient.BeginCreateOrUpdate(ctx, resourceGroupName, applicationSecurityGroupName, parameters, nil)
 	if err != nil {
 		return appsg, fmt.Errorf("cannot create asg %v, reason: %v", applicationSecurityGroupName, err)
 	}
 
-	err = future.WaitForCompletionRef(ctx, asgClient.Client)
+	res, err := poller.PollUntilDone(ctx, nil)
 	if err != nil {
-		return appsg, fmt.Errorf("cannot get asg create or update future response: %v", err)
+		return appsg, fmt.Errorf("cannot get asg create or update poll response: %v", err)
 	}
+	appsg = res.ApplicationSecurityGroup
 
-	return future.Result(asgClient)
+	return appsg, nil
 }
+
 func (asg *azureAsgWrapperImpl) get(ctx context.Context, resourceGroupName string,
-	applicationSecurityGroupName string) (network.ApplicationSecurityGroup, error) {
-	return asg.asgAPIClient.Get(ctx, resourceGroupName, applicationSecurityGroupName)
-}
-func (asg *azureAsgWrapperImpl) listComplete(ctx context.Context, resourceGroupName string) ([]network.ApplicationSecurityGroup, error) {
-	listResultIterator, err := asg.asgAPIClient.ListComplete(ctx, resourceGroupName)
+	applicationSecurityGroupName string) (armnetwork.ApplicationSecurityGroup, error) {
+	var appsg armnetwork.ApplicationSecurityGroup
+	res, err := asg.asgAPIClient.Get(ctx, resourceGroupName, applicationSecurityGroupName, nil)
 	if err != nil {
-		return nil, fmt.Errorf("failed to list of asgs for resource-group: %v, reason %v", resourceGroupName, err)
+		return appsg, fmt.Errorf("cannot retrieve asg %v, reason %v", applicationSecurityGroupName, err)
 	}
 
-	var asgs []network.ApplicationSecurityGroup
-	for ; listResultIterator.NotDone(); err = listResultIterator.NextWithContext(ctx) {
+	appsg = res.ApplicationSecurityGroup
+	return appsg, nil
+}
+
+func (asg *azureAsgWrapperImpl) listComplete(ctx context.Context,
```

Comment:
We have separate files for nsg, asg, rule etc.. Can you move related functions in individual files?